tensorflow_datasets/core/features/image_feature.py

```diff
@@ -123,8 +135,8 @@ def _encode_image(self, np_image):
     """Returns np_image encoded as jpeg or png."""
     if not self._runner:
       self._runner = utils.TFGraphRunner()
-    if np_image.dtype != np.uint8:
-      raise ValueError('Image should be uint8. Detected: %s.' % np_image.dtype)
+    if np_image.dtype not in ACCEPTABLE_NP_DTYPES:
```

Comment:
This should match self._dtype. Please use https://www.tensorflow.org/api_docs/python/tf/dtypes/DType#as_numpy_dtype instead. You can remove `ACCEPTABLE_NP_DTYPES ` afterwards